ISM43362/ISM43362.cpp

```diff
@@ -196,6 +196,61 @@ bool ISM43362::dhcp(bool enabled)
     return (_parser.send("C4=%d", enabled ? 1 : 0) && check_response());
 }
 
+/*
+  * @brief  Update modem_country_code.
+  * @param  country_codeP: User's country code.
+  * @retval boolean value : false ->error
+
+WiFi modem is only able to receive 4 country codes: CA, US, FR and JP.
+CA, US -> 11 channels
+FR     -> 13 channels
+JP     -> 14 channels
+
+For each input country code(country_codeP), function is specifying
+the number of channels : 11, 13 or 14.
+Function is updating private variable modem_country_code.
+
+For instance, country_codeP=VI, function updates modem_country_code to US.
+*/
+
+bool ISM43362::check_country_code(const char* country_codeP)
+{
+  if ((!strcmp(country_codeP, "CA")) || (!strcmp(country_codeP, "US"))
+      || (!strcmp(country_codeP, "FR")) || (!strcmp(country_codeP, "JP")) )
+  {
+     strcpy(modem_country_code, country_codeP);
+    return true;
+  }
+
+  int k = 0;
+
+  while(CountryCodeThirteenChannels[k] != "END")
+  {
+    if (CountryCodeThirteenChannels[k] == country_codeP)
```

Comment:
Are we comparing 2 pointers here ? why not using strcmp ?

Reply:
Code has been updated.